src/seraphis_main/scan_balance_recovery_utils.cpp

```diff
@@ -423,41 +421,48 @@ bool try_find_legacy_enotes_in_tx(const rct::key &legacy_base_spend_pubkey,
         found_an_enote = true;
     }
 
-    // 4. check if there is a main enote ephemeral pubkey
-    if (legacy_main_enote_ephemeral_pubkey == rct::rct2pk(rct::I))
+    // 3. check if there is a main enote ephemeral pubkey
+    if (legacy_main_enote_ephemeral_pubkeys.front() == rct::rct2pk(rct::I))
```

Comment:
This looks like another mismatch to wallet2. wallet2 skips processing the transaction's enotes entirely if there is no main enote ephemeral pub key:

https://github.com/monero-project/monero/blob/ac02af92867590ca80b2779a7bbeafa99ff94dcb/src/wallet/wallet2.cpp#L2285-L2294

Can be handled in a follow-up PR, but I would suggest `extract_legacy_enote_ephemeral_pubkeys_from_tx_extra` can just return false if there is no main enote ephemeral pub key (don't even add `rct::I`), and  `try_find_legacy_enotes_in_tx` can also return false immediately in that case too (and not process additional tx pub keys).

Reply:
To be clear, this comment is totally unrelated to this PR and this looks like a distinct issue already present :)

Reply:
Ya I was debating this. Fine with me to keep it

Reply:
Biggest issue I can think of is you could theoretically send dust to someone who confirms / denies receipt in order to see if they're running a Seraphis lib wallet versus wallet2